src/main/kotlin/org/jetbrains/bio/big/Bed.kt

```diff
@@ -66,67 +78,109 @@ data class BedEntry(
             omitEmptyStrings: Boolean = false
     ): ExtendedBedEntry {
 
-        check(fieldsNumber in 3..12) { "Fields number expected 3..12, but was $fieldsNumber" }
+        check(fieldsNumber in 3..12) { "Fields number expected in range 3..12, but was $fieldsNumber" }
 
-        // This impl parsed only BED9 and adds 9..12 fields to 'rest' string
         val limit = fieldsNumber.toInt() - 3 + 1
-        val it = when {
-            rest.isEmpty() -> emptyArray<String>().iterator()
-            omitEmptyStrings -> Splitter.on(delimiter).trimResults().omitEmptyStrings().limit(limit).split(rest).iterator()
-            else -> rest.split(delimiter, limit = limit).iterator()
+        val fields = when {
+            rest.isEmpty() -> emptyList<String>()
+            omitEmptyStrings -> Splitter.on(delimiter).trimResults().omitEmptyStrings().limit(limit).split(rest)
+            else -> rest.split(delimiter, limit = limit)
+        }.toList()
```

Comment:
OK, then how about we:
1. replace the third branch with another Splitter chain call (otherwise there's not much sense in demanding an iterator, since a list will be created in most cases anyway);
2. switch to an iterator to prevent the unnecessary list generation, and
3. check `hasNext` for each separate field.

This will require nine more `throw` points, but we'll probably get improved performance over my list-based version.

Reply:
We could at least convert 2nd to list, and leave 1st and 3rd as is (already lists). I thing if we use list-based version there is no much sense to convert it to iterator and use `iter.next()` instead of `list[idx]`

Reply:
Done.